### ISSUE TITLE
Moving to segments in table in grid view only works for leftmost image

### ISSUE THREAD

**Issue by constantinpape** [mobie/mobie-viewer-fiji#893]
Moving to segments in a table in a (merged) grid only works for the leftmost image, for all other images it will move to the leftmost image instead of the correct position in the current image.
To reproduce: `covid-if-project`->default->cell_segmentation

It works in the leftmost image:

https://user-images.githubusercontent.com/4263537/199785967-67d51f50-9b78-4e1c-aa5d-a0aa91c0065d.mp4

But fails otherwise:

https://user-images.githubusercontent.com/4263537/199785990-d063d7d9-a96e-4276-b2f8-95d10de62bb4.mp4

**Comment by constantinpape** [comment]
Ok, I don't really know what's going on, for another project (`htm-test`->`nucleus-segmentation`) it doesn't work for the leftmost image either and the table navigation is completely off:

https://user-images.githubusercontent.com/4263537/199787907-8e91d2e4-4aa9-473a-b74f-61aa4b084d40.mp4

**Comment by tischi** [comment]
OK... the selection seems to work properly. 
Then this means that the transformations are not properly applied to the tables.
What is funny is that they are not completely off, but stay within the well...that's the case right?

**Comment by constantinpape** [comment]
> OK... the selection seems to work properly.

Yep, It works. I was just confused there because of the random navigation.

> Then this means that the transformations are not properly applied to the tables.

Yep.

> What is funny is that they are not completely off, but stay within the well...that's the case right?

Looks like it here. But I am not quite sure if it's just a coincidence or if it means that the transformations are applied correctly at the well level, but not at the site level.

**Comment by tischi** [comment]
I started to have a look but I am a bit lost in terms of how to debug this...

Do you we have a view where we just have one simple merged grid with segmented nuclei, without the well/site nesting?

**Comment by tischi** [comment]
Note to self:  For some weird reason, during the initialization it is not only transforming the reference source: `segmentation-nuclei-C01_0` but also some random other source, e.g. `segmentation-nuclei-C03_6`. The latter changes every time I run it. This smells like a bug..

**Comment by constantinpape** [comment]
> Do you we have a view where we just have one simple merged grid with segmented nuclei, without the well/site nesting?

I will check and go ahead and create one if not.

**Comment by tischi** [comment]
Note to self: It seems that only the well translations are applied on the table level:

```
ATM: Updating transforms of segmentation-nuclei-C01_6: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_7: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_1: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_8: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_2: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_3: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_4: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C01_5: 3d-affine: (1.0, 0.0, 0.0, 244.4000000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_7: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_6: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_8: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_2: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_1: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_3: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_4: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C02_5: 3d-affine: (1.0, 0.0, 0.0, 3135.6000000000004, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_6: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_7: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_8: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_1: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_2: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_3: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_4: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
ATM: Updating transforms of segmentation-nuclei-C03_5: 3d-affine: (1.0, 0.0, 0.0, 6026.800000000001, 0.0, 1.0, 0.0, 5486.000000000001, 0.0, 0.0, 1.0, 0.0)
```

However, then I would expect that all the position would be in the upper left corner of the well, which is not the case as it moves around more.

@constantinpape are you sure that in this data set the nuclei positions in the tables are in calibrated units and not mistakenly in voxel units? It actually feels like there is a factor of two too much in the tables.

**Comment by tischi** [comment]
Here, I am talking about `"https://github.com/mobie/htm-test"` `"simple-grid"`.

**Comment by constantinpape** [comment]
> @constantinpape are you sure that in this data set the nuclei positions in the tables are in calibrated units and not mistakenly in voxel units? It actually feels like there is a factor of two too much in the tables.

Yeah, maybe this is wrong. I will double check as well.

**Comment by tischi** [comment]
Thanks!
Two simultaneous bugs leading to confusing behavior ;-)
I am off for today....

**Comment by constantinpape** [comment]
> Thanks!
> Two simultaneous bugs leading to confusing behavior ;-)

Yes, that might be the case.

> I am off for today....

I will check / create the new view by tomorrow.

**Comment by constantinpape** [comment]
Hey @tischi,
- you we're right, something was wrong with the coordinates in the tables! I divided by the resolution instead of multiplying... This is fixed now in the `htm-test` repo. (Two other projects are affected by this as well, I will fix this now as well).
- I created the view `single_well`, which contains the nuclei and nucleus segmentation for a single well with a single grid transform.

Now that the coordinate issue is fixed in the table the bug on the Fiji side also looks much clearer:
- Everything in `single_well` works as expected.
- in the `default` view with multiple wells everything works in the first well, but it doesn't work in the other wells. In the cases the trasformation always goes to the first well, so it looks like the well level transformation is just not applied to the table.

**Comment by constantinpape** [comment]
Actually the other projects I thought would be affected were not (one of them had a pixel size of 1, and for the other, `covid-if`, I used some custom code that did not have the bug.)
I am just mentioning this because it also explains why we've seen inconsistent behavior between the `htm-test` project and `covid-if` when trying to debug this.

**Comment by tischi** [comment]
thanks for fixing the anchor points.

> in the default view with multiple wells everything works in the first well, but it doesn't work in the other wells. In the cases the trasformation always goes to the first well, so it looks like the well level transformation is just not applied to the table.

maybe you had some "typos", but for me it is rather: for _all_ wells in the `default` view the `sites` transformations are not applied, thus one always stays in the top left `site`.

**Comment by tischi** [comment]
Should be fixed with `3.0.3-SNAPSHOT`.

**Comment by constantinpape** [comment]
Thanks, it's fixed now.
